.travis.yml

```diff
@@ -123,13 +132,24 @@ script:
     - sudo docker build -t $DOCKER_REPOSITORY .
 
     # Build acceptance tests container
-    - ./tests/build-acceptance ./tests ./docs/internal_api.yml
+    - if [ "$JOB_TYPE" = compile_and_basic_tests ]; then ./tests/build-acceptance ./tests ./docs/internal_api.yml; fi
 
     # Clone integration repo. for api testing (purposely after license checking..)
     - git clone https://github.com/mendersoftware/integration.git
 
     # Run dependant microservices and the tests service itself
-    - TESTS_DIR=$PWD/tests ./tests/run-acceptance $PWD/integration ./tests/docker-compose.yml
+    - if [ "$JOB_TYPE" = compile_and_basic_tests ]; then
+        TESTS_DIR=$PWD/tests ./tests/run-test-environment "acceptance" $PWD/integration ./tests/docker-compose.yml;
+      fi
+
+    # Run dependant microservices and the tests service itself
+    - if [ "$JOB_TYPE" = compile_and_integration_tests ]; then
+        NO_CLEANUP=true TESTS_DIR=$PWD/tests ./tests/run-test-environment "integration" $PWD/integration ./tests/docker-compose-integration.yml &
+      fi
+
+    - if [ "$JOB_TYPE" = compile_and_integration_tests ]; then
+        cd $PWD/integration/tests && bash run.sh;
```

Comment:
I'd put parentheses around this to avoid being in the wrong folder later on if we add any steps here:

```
( cd $PWD/integration/tests && bash run.sh )
```